src/Analyzer.Reports/ConsoleReportWriter.cs

```diff
@@ -17,61 +16,59 @@ public class ConsoleReportWriter : IReportWriter
         internal static string IndentedNewLine = Environment.NewLine + "\t";
         internal static string TwiceIndentedNewLine = Environment.NewLine + "\t\t";
 
+        private readonly List<string> filesAlreadyOutput = new List<string>();
+
         /// <inheritdoc/>
         public void WriteResults(IEnumerable<Types.IEvaluation> evaluations, IFileInfo templateFile, IFileInfo parametersFile = null)
         {
-            // Log info on file to be analyzed
-            string fileMetadata = Environment.NewLine + Environment.NewLine + $"File: {templateFile}";
-            if (parametersFile != null)
+            var resultsByFile = ReportsHelper.GetResultsByFile(evaluations, filesAlreadyOutput, out int passedEvaluations);
+
+            // output files in sorted order, but always output root first and regardless if no results
+            var filesWithResults = resultsByFile.Keys.ToList();
+            filesWithResults.Remove(templateFile.FullName);
```

Comment:
Can there be duplicates in this list?

Reply:
The list is from the keys of a dictionary so it should not have duplicates.